scripts/tso-brute.nse

```diff
@@ -94,32 +94,47 @@ Driver = {
 
     local commands = self.options['key1']
     local always_logon = self.options['key2']
+    local skip = self.options['skip']
     stdnse.debug(2,"Getting to TSO")
     local run = stdnse.strsplit(";%s*", commands)
+    stdnse.verbose(2,"Trying User ID/Password: %s/%s", user, pass)
     for i = 1, #run do
-      stdnse.debug(2,"%s: Issuing Command (#%s of %s): %s",user , i, #run ,run[i])
-      self.tn3270:send_cursor(run[i])
+      stdnse.debug(2,"Issuing Command (#%s of %s): %s", i, #run ,run[i])
+      if i == #run and run[i]:upper():find("LOGON APPLID") and skip then
+        stdnse.debug(2,"Trying User ID: %s", user)
+        self.tn3270:send_cursor(run[i] .. " DATA(" .. user .. ")")
+      elseif i == #run and skip then
+        stdnse.debug(2,"Trying User ID: %s", user)
+        self.tn3270:send_cursor(run[i] .. " " .. user)
+      else
+        self.tn3270:send_cursor(run[i])
+      end
       self.tn3270:get_all_data()
     end
 
-    if ( self.tn3270:find('NO USER APPLID AVAILABLE') ) then
-      local err = brute.Error:new( "No APPLID Available" )
-        -- This error occurs on too many concurrent application requests
-        -- it should be temporary.
+    if self.tn3270:find("***") then -- For ACF2/TopSecret if required
+      self.tn3270:send_enter()
+      self.tn3270:get_all_data()
+    end
+
+    if not self.tn3270:find("ENTER USERID") 
+       and not self.tn3270:find("TSO/E LOGON")
+       and not self.tn3270:find("IKJ56710I INVALID USERID") then
+      local err = brute.Error:new( "TSO Unavailable" )
+      stdnse.debug(1,"TSO Unavailable for UserID %s", user )
+        -- This error occurs on too many concurrent application requests it
+        -- should be temporary. If not, the script errors and less threads should be used.
```

Comment:
Wow, you get to use a new feature of brute.lua that @sergeykhegay added last summer! At this point, call [`err:setReduce(true)`](https://nmap.org/nsedoc/lib/brute.html#setReduce) to signal to the brute engine that you should reduce the number of running threads and try again. Check out `ftp-brute.nse` to see this in action.